src/component/FetchItems.js

```diff
@@ -53,9 +53,9 @@ const FetchItems = ({ token, setToken, firestore }) => {
   // based on whether an item has been purchased within
   // the last 24 hours
   const calculateIfPurchased = item => {
-    const dateOfPurchaseJS = dayjs(item.dateOfPurchase.toDate());
-
-    if (today.diff(dateOfPurchaseJS, 'hour') <= 24) {
+    if (item.dateOfPurchase === undefined) {
+      return 'nonPurchasedItem';
+    } else if (today.diff(dayjs(item.dateOfPurchase.toDate()), 'hour') <= 24) {
```

Comment:
This ~was~ is a bit difficult to read, what do you think about moving it to an util function and give it a super meaningful name? If you do it, we can replace this line with something like

```js
else if (wasItemPurchasedToday(today)) {...}
```